From de8078c0f41946c3458aad57ea9b01e4941cda56 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Tue, 8 Nov 2022 15:31:40 +0200
Subject: [PATCH 02/11] [openmp] Only try to add the -mrtm flag on i386/x86_64

Clang does accept the flag for any architecture, but prints a
warning:

    clang-16: warning: argument unused during compilation: '-mrtm' [-Wunused-command-line-argument]

This flag is documented as an x86 specific flag, so don't try to
add it for other architectures, to silence this warning.

Differential Revision: https://reviews.llvm.org/D137744
---
 openmp/runtime/cmake/LibompHandleFlags.cmake | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/openmp/runtime/cmake/LibompHandleFlags.cmake b/openmp/runtime/cmake/LibompHandleFlags.cmake
index 33847b5545fb..2eb9ee640834 100644
--- a/openmp/runtime/cmake/LibompHandleFlags.cmake
+++ b/openmp/runtime/cmake/LibompHandleFlags.cmake
@@ -42,7 +42,9 @@ function(libomp_get_cxxflags cxxflags)
   libomp_append(flags_local /GS LIBOMP_HAVE_GS_FLAG)
   libomp_append(flags_local /EHsc LIBOMP_HAVE_EHSC_FLAG)
   libomp_append(flags_local /Oy- LIBOMP_HAVE_OY__FLAG)
-  libomp_append(flags_local -mrtm LIBOMP_HAVE_MRTM_FLAG)
+  if(${IA32} OR ${INTEL64})
+    libomp_append(flags_local -mrtm LIBOMP_HAVE_MRTM_FLAG)
+  endif()
   # Intel(R) C Compiler flags
   libomp_append(flags_local /Qsafeseh LIBOMP_HAVE_QSAFESEH_FLAG)
   libomp_append(flags_local -Qoption,cpp,--extended_float_types LIBOMP_HAVE_EXTENDED_FLOAT_TYPES_FLAG)
-- 
2.25.1

